vm/src/compile.rs

```diff
@@ -47,6 +44,26 @@ pub fn compile(vm: &mut VirtualMachine, source: &String, mode: Mode) -> Result<P
     ))
 }
 
+pub fn compile_ast(vm: &mut VirtualMachine, ast: ast::Statement) -> PyResult {
+    let mut compiler = Compiler::new();
+    compiler.push_code_object(CodeObject::new(Vec::new()));
+    if let ast::Statement::Expression {ref expression} = ast {
+        compiler.compile_expression(expression);
+    } else {
+        compiler.compile_statement(&ast);
+        compiler.emit(Instruction::LoadConst {
+            value: bytecode::Constant::None,
+        });
+    }
+    compiler.emit(Instruction::ReturnValue);
+
+    let code = compiler.pop_code_object();
+    Ok(PyObject::new(
+        PyObjectKind::Code { code: code },
+        vm.get_type(),
+    ))
+}
+
 pub enum Mode {
     Exec,
```

Comment:
Can you add single mode here?

Reply:
Yeah, that's what I'm planning.

Reply:
Okay, please do this, then the compile_ast can be merged into compile with the single option.